core/app/models/spree/order.rb

```diff
@@ -692,6 +693,18 @@ def after_cancel
       payments.completed.each { |payment| payment.cancel! }
       send_cancel_email
       self.update!
+
+      # Free up authorized store credits
+      payments.store_credits.pending.each { |payment| payment.void! }
```

Comment:
why would we be in a state with pending store credits where we had never previously been concerned about pending payments of another kind?